cli/root.go

```diff
@@ -85,15 +84,46 @@ func NewAlertmanagerClient(amURL *url.URL) *client.Alertmanager {
 
 	cr := clientruntime.New(address, path.Join(amURL.Path, defaultAmApiv2path), schemes)
 
-	cr.Transport = &http.Transport{
-		TLSClientConfig: tlsConfig,
+	if amURL.User != nil && httpConfigFile != "" {
+		kingpin.Fatalf("basic authentication and http.config.file are mutually exclusive")
 	}
 
 	if amURL.User != nil {
 		password, _ := amURL.User.Password()
 		cr.DefaultAuthentication = clientruntime.BasicAuth(amURL.User.Username(), password)
 	}
 
+	var httpConfig *promconfig.HTTPClientConfig
+	if httpConfigFile != "" {
+		var err error
+		httpConfig, err = config.LoadHttpConfig(httpConfigFile)
+		if err != nil {
+			kingpin.Fatalf("could not load http config file: %v\n", err)
+		}
+
+		tlsConfig, err := promconfig.NewTLSConfig(&httpConfig.TLSConfig)
+		if err != nil {
+			kingpin.Fatalf("failed to create tls config: %v\n", err)
+		}
+
+		cr.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+
+		if httpConfig.BasicAuth != nil {
+			password, _ := config.FetchBasicAuthPassword(httpConfig.BasicAuth)
+			cr.DefaultAuthentication = clientruntime.BasicAuth(httpConfig.BasicAuth.Username, password)
+		}
+
+		if httpConfig.Authorization.Type == "Bearer" {
```

Comment:
To be consistent with what the HTTP configuration file exposes, we shouldn't limit authorization type to bearer token only. The same is true for other settings like OAuth2 and proxy.
It would be better to create a custom HTTP client from the given configuration and use `clientruntime.NewWithClient()`?

Reply:
@simonpasquier Thanks for your comments. I would like to have this PR to cover bearer token firstly. I will have a following PR to address a custom HTTP client for other cases. Does it make sense? we have customers to wait for bearer token support.

Reply:
I'd prefer to have everything in the same PR otherwise it may confuse our users.

Reply:
@simonpasquier PTAL. Thanks.